app/code/community/Bolt/Boltpay/Helper/Api.php

```diff
@@ -165,121 +165,141 @@ public function verify_hook($payload, $hmacHeader)
+            // setting Bolt as payment method
+            $immutableQuote->getShippingAddress()->setPaymentMethod(Bolt_Boltpay_Model_Payment::METHOD_CODE)->save();
+            $payment = $immutableQuote->getPayment();
+            $payment->setMethod(Bolt_Boltpay_Model_Payment::METHOD_CODE);
+
+            // adding transaction data to payment instance
+            $payment->setAdditionalInformation('bolt_transaction_status', $transactionStatus);
+            $payment->setAdditionalInformation('bolt_reference', $reference);
+            $payment->setAdditionalInformation('bolt_merchant_transaction_id', $transaction->id);
+            //add if called by ajax request, this info would be used for log
+            $payment->setAdditionalInformation('is_ajax_request', $isAjaxRequest);
+            $payment->setTransactionId($transaction->id);
+
+            Mage::helper('boltpay')->collectTotals($immutableQuote, true)->save();
+
+            // a call to internal Magento service for order creation
+            $service = Mage::getModel('sales/service_quote', $immutableQuote);
+
+            try {
+                $service->submitAll();
+            } catch (Exception $e) {
+                Mage::helper('boltpay/bugsnag')->addBreadcrumb(
+                    array(
+                        'transaction'   => json_encode((array)$transaction),
+                        'quote_address' => var_export($immutableQuote->getShippingAddress()->debug(), true)
+                    )
                 );
-                Mage::helper('boltpay/bugsnag')->notifyException(new Exception($errorMessage), $metaData);
+                throw $e;
             }
-        }
-
-        // setting Bolt as payment method
-        $immutableQuote->getShippingAddress()->setPaymentMethod(Bolt_Boltpay_Model_Payment::METHOD_CODE)->save();
-        $payment = $immutableQuote->getPayment();
-        $payment->setMethod(Bolt_Boltpay_Model_Payment::METHOD_CODE);
 
-        // adding transaction data to payment instance
-        $payment->setAdditionalInformation('bolt_transaction_status', $transactionStatus);
-        $payment->setAdditionalInformation('bolt_reference', $reference);
-        $payment->setAdditionalInformation('bolt_merchant_transaction_id', $transaction->id);
-        //add if called by ajax request, this info would be used for log
-        $payment->setAdditionalInformation('is_ajax_request', $isAjaxRequest);
-        $payment->setTransactionId($transaction->id);
-
-        Mage::helper('boltpay')->collectTotals($immutableQuote, true)->save();
-      
-        // a call to internal Magento service for order creation
-        $service = Mage::getModel('sales/service_quote', $immutableQuote);
+        } catch ( Exception $e ) {
+            // Order creation failed, so mark the parent quote as active so webhooks can retry it
+            if ($parentQuote) {
+                $parentQuote->setIsActive(true);
```

Comment:
Should we call save here so that setting it to active will be restored in the database?

Reply:
good call.